src/main/kotlin/no/skatteetaten/aurora/fergus/service/StorageGridService.kt

```diff
@@ -93,17 +97,29 @@ class StorageGridServiceReactive(
         val displayGroupName = createDisplayGroupName(bucketName, path, access)
 
         // Get group by shortname
-        val getGroupResponse = storageGridGroupsApi
-            .orgGroupsGroupShortNameGet(shortGroupName)
-            .awaitSingle()
-        if (getGroupResponse.status === GetPatchPostPutGroupResponse.StatusEnum.ERROR) {
-            throw ResponseStatusException(
-                HttpStatus.INTERNAL_SERVER_ERROR,
-                "The Storagegrid groups api returned an error on orgGroupsGroupShortNameGet"
-            )
+        var getGroupResponse: GetPatchPostPutGroupResponse?
```

Comment:
Man prøver ofte å unngå bruken av var i kotlin. Et forslag:
```
        // Get group by shortname
        val getGroupResponse = try {
            storageGridGroupsApi
                .orgGroupsGroupShortNameGet(shortGroupName)
                .awaitSingle()
        } catch (wcre: WebClientResponseException) {
            if (wcre.statusCode == HttpStatus.NOT_FOUND) {
                logger.info("Got NOT_FOUND when trying to get named group, which is normal")
                null
            } else {
                logger.error("Error trying to get existing group", wcre)
                throw wcre
            }
        }

        if (getGroupResponse != null && getGroupResponse.status === GetPatchPostPutGroupResponse.StatusEnum.ERROR) {
            throw ResponseStatusException(
                HttpStatus.INTERNAL_SERVER_ERROR,
                "The Storagegrid groups api returned an error on orgGroupsGroupShortNameGet"
            )
        }

```

Reply:
Jeg forsøkte noen varianter her, men fikk ikke denne til uten var. Skal se om ditt forslag fungerer.

Reply:
Det går an å dra det enda lenger siden denne typen koden eksisterer to plasser allerede, usikker om det er verdt det. Du ser hva du foretrekker
```
       suspend fun <T> tryOrNullIfNotFound(operation: String, fn: suspend () -> T): T? = try {
        fn()
    } catch (wcre: WebClientResponseException) {
        if (wcre.statusCode == HttpStatus.NOT_FOUND) {
            logger.info("Got NOT_FOUND when trying to get $operation, which is normal")
            null
        } else {
            logger.error("Error trying to get $operation", wcre)
            throw wcre
        }
    }
```

Du kan da bruke den slik
```
val getGroupResponse = tryOrNullIfNotFound("existing group") {
            storageGridGroupsApi
                .orgGroupsGroupShortNameGet(shortGroupName)
                .awaitSingle()
        }
```